### ISSUE TITLE
[BUG] New-PnPSite: Object reference not set to an instance of an object. (GCC High)

### ISSUE THREAD

**Issue by jrudley** [pnp/powershell#2948]
### Notice
Many bugs reported are actually related to the PnP Framework which is used behind the scenes. Consider carefully where to report an issue:

1. **Are you using ```Invoke-PnPSiteTemplate``` or ```Get-PnPSiteTemplate```**? The issue is most likely related to the Provisioning Engine. The Provisioning engine is _not_ located in the PowerShell repo. Please report the issue here: https://github.com/pnp/pnpframework/issues.
1. **Is the issue related to the cmdlet itself, its parameters, the syntax, or do you suspect it is the code of the cmdlet that is causing the issue?** Then please continue reporting the issue in this repo.
1. **If you think that the functionality might be related to the underlying libraries that the cmdlet is calling** (We realize that might be difficult to determine), please first double check the code of the cmdlet, which can be found here: https://github.com/pnp/powershell/tree/master/src/Commands. If related to the cmdlet, continue reporting the issue here, otherwise report the issue at https://github.com/pnp/pnpframework/issues

### Reporting an Issue or Missing Feature
new-pnpsite throws object reference not set to an instance of an object

### Expected behavior 
new modern team site created

### Actual behavior 
Create new modern Team Site

### Steps to reproduce behavior
Please include complete script or code samples in-line or linked from [gists](https://gist.github.com/)
 Connect-PnPOnline -AzureEnvironment USGovernmentHigh  -ClientId 'xxx' -Url 'https://x.sharepoint.us' -CertificatePassword $pass  -Tenant x.onmicrosoft.com -CertificatePath C:\mycertificates\x.pfx

New-PnPSite -Type TeamSite -Title 'Team Contoso' -Alias contoso
New-PnPSite: Object reference not set to an instance of an object.

### What is the version of the Cmdlet module you are running?
(you can retrieve this by executing ```Get-Module -Name "PnP.PowerShell" -ListAvailable```)
Manifest   2.0.53     PnP.PowerShell

### Which operating system/environment are you running PnP PowerShell on?
- [ X] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by gautamdsheth** [comment]
You also need to set owners for the team site, to do that, please specify value for `-Owners` parameters, can you please do that, execute and let us know ?

**Comment by jrudley** [comment]
> You also need to set owners for the team site, to do that, please specify value for `-Owners` parameters, can you please do that, execute and let us know ?

Hi,
I kept the example simple from the documentation. 
![image](https://user-images.githubusercontent.com/1934411/229133488-6f7d6d05-2877-4635-b15e-90ce8c14d874.png)

New-PnPSite -Type TeamSite -Title 'Team Contoso' -Alias contoso -Owners @($SiteCollectionAdmin1, $SiteCollectionAdmin2)
New-PnPSite: Object reference not set to an instance of an object.

I added owners parameter with the same issue. My original command is below which also has the same issue.

 New-PnPSite -Type TeamSite -Title $NewSiteTitle -Alias $NewSiteAlias -Description $NewSiteDescription -Owners @($SiteCollectionAdmin1, $SiteCollectionAdmin2)

**Comment by jrudley** [comment]
I debugged this a bit more using fiddler. I authenticate in and make my call. The users call is going correctly, but the groups call is going to azure commercial instead of azure gov.
![image](https://user-images.githubusercontent.com/1934411/229145714-0dd92f37-8be1-4728-890d-db23d5502922.png)

![image](https://user-images.githubusercontent.com/1934411/229146260-20a134fb-2433-4816-a01f-460f6a842c72.png)

**Comment by addphanik** [comment]
I am also facing the same issue. I have added -Owners as well. I am using ManagedIdentity for authentication in Azure Function. PnPPowerShell Version - 2.1.1 PowerShell: 7.2

Is it correct way to pass owners parameter ? -Owners "admin@mytenant.onmicrosoft.com"

I was able to create of type "TeamSiteWithoutOfficegroup" though.

New-PnPSite -Type TeamSite -Description "Test project site" -PreferredDataLocation "CAN" -Title $siteTitle -Alias "ESU-PRJ-$($requestID)" -SiteAlias "ESU-PRJ-$($requestID)" -TimeZone UTCPLUS0200_HELSINKI_KYIV_RIGA_SOFIA_TALLINN_VILNIUS -Owners "admin@mytenant.onmicrosoft.com" -Lcid 1033 -Connection $conn -Wait -Classification "HBI"

**Comment by addphanik** [comment]
> I am also facing the same issue. I have added -Owners as well. I am using ManagedIdentity for authentication in Azure Function. PnPPowerShell Version - 2.1.1 PowerShell: 7.2
> 
> Is it correct way to pass owners parameter ? -Owners "[admin@mytenant.onmicrosoft.com](mailto:admin@mytenant.onmicrosoft.com)"
> 
> I was able to create of type "TeamSiteWithoutOfficegroup" though.
> 
> New-PnPSite -Type TeamSite -Description "Test project site" -PreferredDataLocation "CAN" -Title siteTitle−Alias"ESU−PRJ−(requestID)"−SiteAlias"ESU−PRJ−($requestID)" -TimeZone UTCPLUS0200_HELSINKI_KYIV_RIGA_SOFIA_TALLINN_VILNIUS -Owners "[admin@mytenant.onmicrosoft.com](mailto:admin@mytenant.onmicrosoft.com)" -Lcid 1033 -Connection $conn -Wait -Classification "HBI"

ERROR: Object reference not set to an instance of an object. Exception : Type : System.Management.Automation.PSInvalidOperationException ErrorRecord : Exception : Type : System.Management.Automation.ParentContainsErrorRecordException Message : Object reference not set to an instance of an object. HResult : -2146233087 CategoryInfo : InvalidOperation: (:) [], ParentContainsErrorRecordException FullyQualifiedErrorId : InvalidOperation TargetSite : Name : ProcessRecord DeclaringType : PnP.PowerShell.Commands.Base.PnPConnectedCmdlet MemberType : Method Module : PnP.PowerShell.dll Message : Object reference not set to an instance of an object. Source : PnP.PowerShell HResult : -2146233079 StackTrace : at PnP.PowerShell.Commands.Base.PnPConnectedCmdlet.ProcessRecord() in c:\build\src\Commands\Base\PnPConnectedCmdlet.cs:line 98 at PnP.PowerShell.Commands.PnPSharePointCmdlet.ProcessRecord() in c:\build\src\Commands\Base\PnPSharePointCmdlet.cs:line 120 at System.Management.Automation.Cmdlet.DoProcessRecord() at System.Management.Automation.CommandProcessor.ProcessRecord() CategoryInfo : InvalidOperation: (:) [New-PnPSite], PSInvalidOperationException FullyQualifiedErrorId : InvalidOperation,PnP.PowerShell.Commands.NewSite InvocationInfo : MyCommand : New-PnPSite ScriptLineNumber : 27 OffsetInLine : 1 HistoryId : 1 ScriptName : C:\home\site\wwwroot\HttpTrigger1\run.ps1 Line : New-PnPSite -Type TeamSite -Description "Test project site" -PreferredDataLocation "CAN" -Title $siteTitle -Alias "ESU-PRJ-$($requestID)" -SiteAlias "ESU-PRJ-$($requestID)" -TimeZone UTCPLUS0200_HELSINKI_KYIV_RIGA_SOFIA_TALLINN_VILNIUS -Owners "admin@mytenent.onmicrosoft.com" -Lcid 1033 -Wait -Classification "HBI" -Connection $conn

**Comment by jrudley** [comment]
@gautamdsheth Any potential updates?

**Comment by gautamdsheth** [comment]
hi @jrudley , can you give it a try with the latest build, 2.1.65-nightly or later ? We found some places where we missed passing the correctly URLs.

Unfortunately, we dont have access to USGov tenants, so would be great if you could test it out for us.

**Comment by jrudley** [comment]
Sure, I will test it out today and let you know.

**Comment by jrudley** [comment]
great news, it works! Thanks again for the fix.

**Comment by gautamdsheth** [comment]
Thanks for confirming the fix and raising the issue. 
Much appreciated, closing this.

**Comment by AlyaKoni** [comment]
Problem still exists running in VS-Code!

```
VS-Code
Version: 1.78.2 (system setup)
Commit: b3e4e68a0bc097f0ae7907b217c1119af9e03435
Date: 2023-05-10T14:39:26.248Z
Electron: 22.5.2
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: Yes
```

```
get-Module PnP.PowerShell
ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Manifest   2.1.69     nightly    PnP.PowerShell                      {Add-PnPAlert, Add-PnPApp, Add-PnPApplicationCustomizer, Add-PnPAvailableSiteClassification…}
```

```
$PSVersionTable
Name                           Value
----                           -----
PSVersion                      7.3.4
PSEdition                      Core
GitCommitId                    7.3.4
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

```
New-PnPSite -Connection $adminCon -Type "TeamSite" -Title $title -Alias "$($siteUrl)" -Description $description -Owners $siteOwners -Wait

New-PnPSite: C:\Alya\Repos\STHODO-ADM-CloudConfiguration\scripts\sharepoint\Create-Site.ps1:145:17
Line |
 145 |  …     $site = New-PnPSite -Connection $adminCon -Type "TeamSite" -Title …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Object reference not set to an instance of an object.
```

**Comment by jrudley** [comment]
Can you run fiddler and see the calls? I don't have an issue. I am working
with Microsoft on a similar issue but it is using a group to create a
aharepoint site and not pnp.

**Comment by AlyaKoni** [comment]
I see the login request:
```
$AlyaConnection = Connect-PnPOnline -Url $TenantAdminUrl -ReturnConnection -Interactive

CONNECT tenantname.sharepoint.com:443 HTTP/1.1
Host: tenantname.sharepoint.com:443
User-Agent: NONISV|SharePointPnP|PnPPS/2.1.1.0 (Microsoft Windows NT 10.0.19045.0)
```

I see the test, if the site exists:
```
$site = Get-PnPTenantSite -Connection $AlyaConnection  -Url $absSiteUrl -ErrorAction SilentlyContinue

POST https://tenantname.sharepoint.com/_vti_bin/client.svc/ProcessQuery HTTP/1.1
Host: tenantname.sharepoint.com
X-ClientService-ClientTag: PnPPS:2.1:Get-PnPTenantSite
Authorization: Bearer **********
User-Agent: NONISV|SharePointPnP|PnPPS/2.1.1.0 (Microsoft Windows NT 10.0.19045.0)
Connection: Keep-Alive
Accept-Encoding: gzip, deflate
Content-Type: text/xml
Content-Length: 729
```

May then, NOTHING. The following line does not generate any request:
```
$site = New-PnPSite -Connection $adminCon -Type "TeamSite" -Title $title -Alias "$($siteUrl)" -Description $description -Owners $siteOwners -Wait
```

**Comment by bed428** [comment]
Also running into this same issue in GCC (VSCode and PS 7.4.5), but with a different commandlet: 

`
$Batch = New-PnPBatch
foreach($item in $items){
     Set-PnPListItem -Identity $item.id -List $List -ContentType "ContentType" -UpdateType SystemUpdate -Values @{"Field 1" = $Value1; "Field 2" = $Value2; "Field 3" = $Value3; "Field 4" = $Value4} -Batch $Batch
}
`
Tried also: 
`
$Batch = New-PnPBatch
$items | Foreach-Object{ Set-PnPListItem -List $List -ContentType "ContentType" -UpdateType SystemUpdate -Values @{"Field 1" = $Value1; "Field 2" = $Value2; "Field 3" = $Value3; "Field 4" = $Value4} -Batch $Batch}
`
And: 
`
$Batch = New-PnPBatch
$items | Foreach-Object{ Set-PnPListItem $_.Id -List $List -ContentType "ContentType" -UpdateType SystemUpdate -Values @{"Field 1" = $Value1; "Field 2" = $Value2; "Field 3" = $Value3; "Field 4" = $Value4} -Batch $Batch}
`
The Set-PnP command works in both without the $Batch.

Potentially the same issue that was fixed in 2.1.65-nightly by @gautamdsheth ?
